src/Extension/CoreExtension.php

```diff
@@ -1285,6 +1287,75 @@ public static function testEmpty($value): bool
         return '' === $value || false === $value || null === $value || [] === $value;
     }
 
+    /**
+     * Checks if a variable is traversable.
+     *
+     *    {# evaluates to true if the foo variable is an array or a traversable object #}
+     *    {% if foo is iterable %}
+     *        {# ... #}
+     *    {% endif %}
+     *
+     * @param mixed $value A variable
+     *
+     * @return bool true if the value is traversable
+     *
+     * @deprecated since Twig 3.8, to be removed in 4.0 (use the native "is_iterable" function instead)
+     *
+     * @internal
+     */
+    public static function testIterable($value)
+    {
+        return is_iterable($value);
+    }
+
+    /**
+     * Checks if a variable is a sequence.
+     *
+     * Traversables must be converted before being tested.
```

Comment:
this behavior is not consistent with the behavior of other Twig APIs, and I don't think we expose the `toArray` method to Twig code, so it is probably impossible to satisfy this requirement easily.

Reply:
@dunglas can we talk about this part? I may have misunderstood an advice shared in Rennes mate March

Reply:
Have a look at the `CoreExtension::toArray()` code, it is used everywhere in the file to easily convert a sequence to an array when possible so that this operation is transparent for users.

Reply:
I didn't know iterator_to_array

Reply:
Done in https://github.com/twigphp/Twig/pull/3859/commits/95f6983b671cfbf2d8b0ed096de57cbdbc0e172f  

Two tests use [ArrayIterator](https://www.php.net/manual/fr/class.arrayiterator.php) which implements [SeekableIterator](https://www.php.net/manual/fr/class.seekableiterator.php) which extends [Traversable](https://www.php.net/manual/fr/class.traversable.php) (through [Iterator](https://www.php.net/manual/fr/class.iterator.php)) and they are still passing OK